### DISCUSSION TITLE
Why do Strings implement the The STRING_DISPLAY protocol? Why not just stick with traits?

### DISCUSSION THREAD

**Discussion by J-Cake** [rune-rs/rune#628] [General]
what happened to `trait`s? Is this a technical thing, or do protocols fill this gap?

**Comment by udoprog** [comment]
Protocols are the stop-gap solution to what will eventually be traits in Rune. They're fairly similar to Python's `__hash__` or `__eq__` methods but are currently not nameable in Rune itself (without calling a native method).

Without type information trait methods are somewhat hard to use, so proper traits are probably not something that will land until we've started working on gradual typing.

**Reply by J-Cake** [reply]
Protocols = _pre_ -trait? That would explain quite a bit actually
